contracts/src/libraries/upgradeable.cairo

```diff
@@ -0,0 +1,27 @@
+
+mod Upgradeable {
+    use zeroable::Zeroable;
+
+    use starknet::syscalls::replace_class_syscall;
+    use starknet::class_hash::ClassHash;
+    use starknet::class_hash::ClassHashZeroable;
+
+    use chainlink::libraries::ownable::Ownable;
+
+    #[event]
+    fn Upgraded(implementation: ClassHash) {}
+
+    fn upgrade_only_owner(impl_hash: ClassHash) {
+        Ownable::assert_only_owner();
+        upgrade(impl_hash);
+    }
```

Comment:
Curious about the choice to bake in the authorization method (Ownable::assert_only_owner()) into the Upgradeable library - would it be better to let the parent contract handle authentication, and simply provide an unprotected `upgrade` function in the library?

Reply:
Alternatively, [OZ's Upgradable library impl](https://github.com/OpenZeppelin/cairo-contracts/pull/603/files#diff-41a9680000506ee92336bbd3f923e942a4e198ae75785c29be3c2def2910d0ea) maintains its own self-contained, simple auth mechanism, which means that there are no dependencies between libraries = less risk, clearer separation of responsibilities (though may not be much of a concern for our use cases?)

Reply:
Yeah this is based on the OZ implementation and I think we should follow the comment here: https://github.com/OpenZeppelin/cairo-contracts/pull/603#pullrequestreview-1406490206

Only exposing the `_upgrade` utility method and the event. That way we can swap this copy out for the OZ library once stabilized.

Reply:
Agreed. This shouldn't be coupled with `Ownable`, and access control should be explicitly required for every upgrade path - added a comment here: https://github.com/smartcontractkit/chainlink-starknet/pull/251#discussion_r1185848788

Reply:
Since we're using the contracts in a very controlled and specific manner, I did it this way to save some code duplication (asserting owner at every contract that uses this library) and that way we'd also only need to test the upgradable.cairo for core logic. I'll change it to be less coupled